scripts/parser/backend/test_cmakecachevariable.py

```diff
@@ -0,0 +1,12 @@
+import unittest
+from cmakecachevariable import cmakecache_variable as cmc_var
+
+class test_cmake_cache_variable(unittest.TestCase):
```

Comment:
That's OK, but one of the things that really needed to be tested here is the parser itself. In particular, the part which extracts all the comments above the variables. I'd like to see a test where, given multi-line strings, they are parsed and the result is checked. For example:

1. case without comments, does it work?
```
A:STRING=val
B:STRING=val2
```
2. case with comments and spaces
```
// this is a comment for variable A
// second line

A:STRING=val

// this is a comment for variable B
B:STRING=val2
```

To achieve this, you will need to refactor the parser class and extract the method which concatenates all the comments above a variable.